lib/contextual_logger.rb

```diff
@@ -133,25 +133,21 @@ def redactor
     end
 
     def format_message(severity, timestamp, progname, message, context: {})
-      message_hash = message_hash_with_context(severity, timestamp, progname, message, context: context)
-
       if @formatter
-        @formatter.call(severity, timestamp, progname, message_hash)
+        @formatter.call(severity, timestamp, progname, { message: ContextualLogger.normalize_message(message), **context })
       else
-        "#{message_hash.to_json}\n"
+        "#{basic_message_formatting(severity, timestamp, progname, message, context: context).to_json}\n"
       end
     end
 
-    def message_hash_with_context(severity, timestamp, progname, message, context:)
-      message_hash =
-        {
-          message:   ContextualLogger.normalize_message(message),
-          severity:  severity,
-          timestamp: timestamp
-        }
+    def basic_message_formatting(severity, timestamp, progname, message, context:)
+      message_hash = {
+        message:   ContextualLogger.normalize_message(message),
+        severity:  severity,
+        timestamp: timestamp
+      }
       message_hash[:progname] = progname if progname
-
-      message_hash.merge!(context)
+      message_hash.merge(context)
```

Comment:
`merge!` is faster since it doesn't have to `dup`. I think that's fine since this method just made a local hash and mutated it one line above.

Reply:
Ah, thanks for clearing that up. I added the comment to the code to indicate the choice so it doesn't get changed again in the future.